### ISSUE TITLE
Cannot monomorphize trait

### ISSUE THREAD

**Issue by Champii** [Champii/Rock#93]
There is a bug with the new struct scope resolution in #91 

Given the following snippet
```haskell
trait Toto a
    toto :: a -> String

impl Toto Int64
    @toto = "1"

impl Toto Bool
    @toto = "true"
```

The following method fails to resolve
```haskell
foo a = a.toto()
```
with
```
thread 'main' panicked at 'BUG: Cannot find function from signature', src/lib/infer/monomorphize/monomorphizer.rs:304:21
```

But it works when using the full-name invocation

```haskell
foo a = Toto::toto a
```

Test main
```haskell
main =
    foo true
    foo 2
```

**Comment by Champii** [comment]
The first polymorphic example on the readme fails seemingly the same way:

```haskell
id a = a

main =
  id(1).print!
  id(2.2).print!
  id("Test").print!
```

**Comment by Champii** [comment]
Fixed in v0.3.0
